common/src/abi/mod.rs

```diff
@@ -50,11 +50,44 @@ pub fn keccak256(x: &[u8]) -> Bytes32 {
     out
 }
 
+pub fn to_eth_signed_message_hash(s: &[u8]) -> Bytes32 {
+    let (bytes, _) = encode_packed(
+        &[
+            Token::String("\x19Ethereum Signed Message:\n".parse().unwrap()),
+            Token::String(s.len().to_string()),
+            Token::Bytes(s.to_vec())
+            ]
+    );
+    keccak256(&bytes)
+}
```

Comment:
Where is this code based from?

Reply:
oh, this code is from openzepplin code base